### ISSUE TITLE
upgrading QGLWidget to QOpenGLWidget

### ISSUE THREAD

**Issue by heary29** [cnr-isti-vclab/meshlab#1259]
Hello, are you considering upgrading QGLWidget to QOpenGLWidget? If so, what should be done? Which aspects do you think are the most important?

**Comment by alemuntoni** [comment]
We are upgrading to QOpenGLWidget, the work is ongoing. There is a a branch called `qopengl` which contains a first non-full-working implementation. Unfortunately, it is not an easy upgrade due to the intense and non-legacy use of QGL classes on meshlab. 
Right now the core of the work has been finished; we need to adapt all the plugins that use GL context to the new QOpenGL classes, and then test everything carefully. 
Hope to have everything working soon!

**Comment by heary29** [comment]
Yes, what you said is great, there are indeed some difficult problems in the replacement, I tried it recently, and the conclusion is the same as yours, looking forward to your release of the new version of QOpenGLWidget, thank you very much for your efforts
